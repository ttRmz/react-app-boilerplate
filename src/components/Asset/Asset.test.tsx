import { fireEvent, render } from "@testing-library/react"

import { Asset, ASSET_NAMES } from "."

describe("<Asset />", () => {
  it("should render an asset", () => {
    const { container } = render(<Asset name={ASSET_NAMES.analysis} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("should pass props on root element", () => {
    const onClick = jest.fn()
    const testId = "asset-id"

    const { container, getByTestId } = render(
      <Asset
        name={ASSET_NAMES.analysis}
        className="test-class"
        id="test-id"
        onClick={onClick}
        data-testid={testId}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(getByTestId(testId))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
