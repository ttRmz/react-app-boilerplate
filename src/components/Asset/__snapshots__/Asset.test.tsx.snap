// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Asset /> should pass props on root element 1`] = `
<figure
  class="Asset test-class"
  data-testid="asset-id"
  id="test-id"
>
  <img
    alt="analysis"
    class="Asset__image"
    src="http://localhost/assets/symbols/analysis.svg"
  />
</figure>
`;

exports[`<Asset /> should render an asset 1`] = `
<figure
  class="Asset"
>
  <img
    alt="analysis"
    class="Asset__image"
    src="http://localhost/assets/symbols/analysis.svg"
  />
</figure>
`;
